.claude/skills/bitwarden-reviewing-changes/SKILL.md

```diff
@@ -0,0 +1,65 @@
+---
+name: bitwarden-reviewing-changes
+description: Claude code review standards for Bitwarden repositories.
+---
+
+# Bitwarden Code Review Skill
+
+## Purpose
+
+Enforce Bitwarden company code review rules to produce concise, consistent, and excellent PR review comments from Claude Code. The instructions are intended to be focused on Claude's behavior when reviewing code, but not specific to any one type of code that we create.
+
+## Rules that **must** be followed
+
+1. **NEVER** use the text label "Issue". **ONLY** list items using the word **Finding**.
+   - Never emit a bare "#" followed immediately by digits (e.g., "#123"); write "Finding 123" instead.
+   - **Findings** only as a numbered list.
+   - **Finding** summary must be a single sentence consisting of less than 30 words.
+
+2. **ALWAYS** consider brevity in the PR summary comment. The summary comment **must not** include detailed references of suggested changes. The finding details and suggested changes **must** be an inline comment in the PR on the precise line for the suggested changes.
+
+3. **ALWAYS** use the Bitwarden Claude Code reviewer guidelines. The type of finding must align with one of our suggested emojis.
+   - ❌ (`:x:`) for a major finding that requires changes
+   - ⚠️ (`:warning:`) for a minor finding that requires a human reviewer's attention
+   - ♻️ (`:recycle:`) for a finding that **creates** technical debt
+   - 🎨 (`:art:`) for a finding that is a significant improvement to the health of the code. **DO NOT** nitpick.
+   - 💭 (`:thought_balloon:`) for a finding that is open inquiry
+
+4. **ALWAYS** check for existing summary comment to avoid duplicate summary comments. If a summary comment does not exist, then only create **ONE** summary comment.
+
+5. **ALWAYS** read the prior summary comment **before** starting on a code review.
+
+6. **ALWAYS** check for existing comment threads **before** starting on a code review to avoid duplicate comments.
+
+7. **NEVER** duplicate a comment thread.
+
+8. **ALWAYS** carefully read all resolved comment threads. You may not reopen a resolved comment thread if the finding is an improvement 🎨 or an inquiry 💭. The human who submitted the PR and the humans that review the PR are ultimately responsible for the consideration and resolution of your suggestions.
+
+9. **NEVER** list the files changed in a pull request in the summary comment. Developers can easily access this information in the Files changes area of a GitHub PR.
+
+10. **ALWAYS** Analyze the changeset systematically:
+
+- What files were modified? (code vs config vs docs)
+- What is the PR title? Does it clearly convey the intent of the code change?
+- What is the PR description? Does it expand upon the PR title to convey important details?
+- Is there new functionality or just modifications?
+- What's the risk level of these changes?
```

Comment:
🎨 I would put this in `<thinking>` blocks around this to enforce structured thinking.


```suggestion
<thinking>
- What files were modified? (code vs config vs docs)
- What is the PR title? Does it clearly convey the intent of the code change?
- What is the PR description? Does it expand upon the PR title to convey important details?
- Is there new functionality or just modifications?
- What's the risk level of these changes?
</thinking>
```